apps/shared/avifpng.c

```diff
@@ -312,6 +307,86 @@ avifBool avifPNGRead(const char * inputFilename,
+        png_uint_32 iccpDataLen = 0;
+        int srgbIntent = 0;
+
+        // PNG specification 4.2.2: The sRGB and iCCP chunks should not both appear.
+        // When the sRGB / iCCP chunk is present applications that recognize it and are capable of color management
+        // must ignore the gAMA and cHRM chunks and use the sRGB / iCCP chunk instead.
+        if (png_get_iCCP(png, info, &iccpProfileName, &iccpCompression, &iccpData, &iccpDataLen) == PNG_INFO_iCCP) {
+            avifImageSetProfileICC(avif, iccpData, iccpDataLen);
+        } else if (png_get_sRGB(png, info, &srgbIntent) == PNG_INFO_sRGB) {
+            // srgb_intent ignored
+            avif->colorPrimaries = AVIF_COLOR_PRIMARIES_BT709;
+            avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_SRGB;
+        } else {
+            avifBool needCraftICC = AVIF_FALSE;
+            double gamma;
+            double wX, wY, rX, rY, gX, gY, bX, bY;
+            float primaries[8];
+            if (png_get_gAMA(png, info, &gamma) == PNG_INFO_gAMA) {
+                gamma = 1.0 / gamma;
+                if (matchesTo3RoundedPlaces(gamma, 2.2)) {
+                    avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_BT470M;
+                } else if (matchesTo3RoundedPlaces(gamma, 1.0)) {
+                    avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_LINEAR;
+                } else if (matchesTo3RoundedPlaces(gamma, 2.8)) {
+                    avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_BT470BG;
+                } else {
+                    needCraftICC = AVIF_TRUE;
+                }
+            } else {
+                gamma = 2.2;
+            }
+
+            if (png_get_cHRM(png, info, &wX, &wY, &rX, &rY, &gX, &gY, &bX, &bY) == PNG_INFO_cHRM) {
+                primaries[0] = (float)rX;
+                primaries[1] = (float)rY;
+                primaries[2] = (float)gX;
+                primaries[3] = (float)gY;
+                primaries[4] = (float)bX;
+                primaries[5] = (float)bY;
+                primaries[6] = (float)wX;
+                primaries[7] = (float)wY;
+                avif->colorPrimaries = avifColorPrimariesFind(primaries, NULL);
+                if (avif->colorPrimaries == AVIF_COLOR_PRIMARIES_UNKNOWN) {
+                    avif->colorPrimaries = AVIF_COLOR_PRIMARIES_UNSPECIFIED;
+                    needCraftICC = AVIF_TRUE;
+                }
+            } else {
+                avifColorPrimariesGetValues(AVIF_COLOR_PRIMARIES_BT709, primaries);
+            }
+
+            if (needCraftICC) {
+                avif->colorPrimaries = AVIF_COLOR_PRIMARIES_UNSPECIFIED;
+                avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_UNSPECIFIED;
+                fprintf(stderr,
+                        "INFO: legacy color management information not matching any CICP value found in file %s. libavif is generating an ICC profile for it."
```

Comment:
color management => color space

"Color space" is the term used in the PNG specification Section 4.2.2.